### ISSUE TITLE
`A Timer is still pending` error when running widget tests

### ISSUE THREAD

**Issue by stevenspiel** [tekartik/sembast.dart#148]
I recently added sembast to my project (thank you for creating/maintaining it), but when I run my tests, I get this error:

```
Pending timers:
Timer (duration: 0:00:00.000100, periodic: false), created:
#0      new _FakeTimer._ (package:quiver/testing/src/async/fake_async.dart:294:42)
#6      Cooperator.cooperate (package:sembast/src/cooperator.dart:24:21)
#7      SembastDatabase.cooperate (package:sembast/src/database_impl.dart:1226:39)
#8      SembastStore.cooperate (package:sembast/src/store_impl.dart:693:36)
#9      SembastStore.txnGetRecord (package:sembast/src/store_impl.dart:479:13)
#10     RecordRefMixin.getSnapshot (package:sembast/src/record_ref_impl.dart:83:10)
#11     RecordRefMixin.get (package:sembast/src/record_ref_impl.dart:73:14)
#12     SembastDBAdapter.get (package:zcommon/services/sembast_db_adapter.dart:18:82)
(...)

══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following assertion was thrown running a test:
A Timer is still pending even after the widget tree was disposed.
```

Is there a way to disable the cooperator for tests? Is that advised?

**Comment by alextekartik** [comment]
I always wondered whether anyone would complain about this weird cooperator concept ;)

I published a version 2.4.3-dev.1 that exports `disableSembastCooperator()` that you should call before running your unit test. (I typically publish a dev version to test using several applications/packages I have in a production context before publishing a release version so it does not mean it is not stable...)

While I write a lot of unit tests, I never managed (or just spent the time) to write relevant flutter unit test (using WidgetTester) with database calls so if your code is public I would be interested in seeing what kind of testing could be done (and whether this solves your issue).

Thanks!

**Comment by stevenspiel** [comment]
Wow, thanks for getting back so quickly! I was able to add `disableSembastCooperator()` to my tests. Unfortunately, the code isn't public and I'm trying to isolate it, but having some trouble.  Regardless, this helped me get on the right path. Thank you!
